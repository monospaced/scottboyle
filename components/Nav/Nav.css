@import url('../../theme/vars.css');

@media (min-width: 639px) {
  .Nav {
    float: left;
  }
}

.Nav-list {
  margin-top: 0;
}

.Nav-heading {
  font-size: var(--pica);
  line-height: 1.5;
  margin-bottom: 0;
}
