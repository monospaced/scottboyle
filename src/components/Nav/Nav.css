@import url("../../theme/vars.css");

@media (--larger-viewport) {
  .Nav {
    float: left;
    text-align: right;
    width: 12.5rem;
  }
}

.Nav-heading {
  font-size: var(--pica);
  line-height: 1.5;
  margin-bottom: 0;
}

.Nav-list {
  margin-bottom: 1.5rem;
  margin-top: 0;
}
