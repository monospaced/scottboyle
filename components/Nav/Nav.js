import React, {Component} from 'react';
import {Link} from 'react-router';
import './Nav.css';

const Nav = ({projects, routes}) => {
  const path = routes[routes.length - 1].path;
  return (
    <nav className="Nav">
      <h2 className="Nav-heading">Work</h2>
      <ul className="Nav-list">
        {Object.keys(projects).map((key) => {
          const title = projects[key].title;
          return (
            <li key={key}>
              {key === path ? title : <Link to={`/${key}`}>{title}</Link>}
            </li>
          );
        })}
        <li>
          {`linklog` === path ? `Linklog` : <Link to={`/linklog/`}>Linklog</Link>}
        </li>
      </ul>
    </nav>
  );
};

module.exports = Nav;
