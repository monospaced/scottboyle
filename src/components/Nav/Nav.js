import { Link } from "react-router";
import React from "react";

import "./Nav.css";

const Nav = ({ projects, routes }) => {
  const path = routes[routes.length - 1].path;

  return (
    <nav className="Nav">
      <h2 className="Nav-heading">Work</h2>
      <ul className="Nav-list">
        {Object.keys(projects).map(key => {
          const title = projects[key].title;

          return (
            <li key={key}>
              {path === key ? title : <Link to={`/${key}`}>{title}</Link>}
            </li>
          );
        })}
        <li>
          {path === `linklog` ? (
            `Linklog`
          ) : (
            <Link to={`/linklog/`}>Linklog</Link>
          )}
        </li>
      </ul>
    </nav>
  );
};

export default Nav;
