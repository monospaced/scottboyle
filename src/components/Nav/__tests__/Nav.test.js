import React from "react";
import { render } from "enzyme";

import data from "../../../scripts/__mocks__/data.js";
import Nav from "../Nav.js";

describe("Nav component", () => {
  const { projects } = data;

  it("should render correctly", () => {
    const component = render(
      <Nav projects={projects} routes={[{ path: "" }]} />,
    );
    expect(component).toMatchSnapshot();
  });

  it("should render a project as the active item", () => {
    const component = render(
      <Nav projects={projects} routes={[{ path: Object.keys(projects)[0] }]} />,
    );
    expect(component).toMatchSnapshot();
  });

  it("should render linklog as the active item", () => {
    const component = render(
      <Nav projects={projects} routes={[{ path: "linklog" }]} />,
    );
    expect(component).toMatchSnapshot();
  });
});
