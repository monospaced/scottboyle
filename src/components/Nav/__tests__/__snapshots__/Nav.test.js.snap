// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav component should render a project as the active item 1`] = `
<nav
  class="Nav"
>
  <h2
    class="Nav-heading"
  >
    Work
  </h2>
  <ul
    class="Nav-list"
  >
    <li>
      Thomson Reuters
    </li>
    <li>
      <a>
        Compass Group
      </a>
    </li>
    <li>
      <a>
        Bilbary
      </a>
    </li>
    <li>
      <a>
        Genes Reunited
      </a>
    </li>
    <li>
      <a>
        The National Lottery
      </a>
    </li>
    <li>
      <a>
        Age Concern
      </a>
    </li>
    <li>
      <a>
        Linklog
      </a>
    </li>
  </ul>
</nav>
`;

exports[`Nav component should render correctly 1`] = `
<nav
  class="Nav"
>
  <h2
    class="Nav-heading"
  >
    Work
  </h2>
  <ul
    class="Nav-list"
  >
    <li>
      <a>
        Thomson Reuters
      </a>
    </li>
    <li>
      <a>
        Compass Group
      </a>
    </li>
    <li>
      <a>
        Bilbary
      </a>
    </li>
    <li>
      <a>
        Genes Reunited
      </a>
    </li>
    <li>
      <a>
        The National Lottery
      </a>
    </li>
    <li>
      <a>
        Age Concern
      </a>
    </li>
    <li>
      <a>
        Linklog
      </a>
    </li>
  </ul>
</nav>
`;

exports[`Nav component should render linklog as the active item 1`] = `
<nav
  class="Nav"
>
  <h2
    class="Nav-heading"
  >
    Work
  </h2>
  <ul
    class="Nav-list"
  >
    <li>
      <a>
        Thomson Reuters
      </a>
    </li>
    <li>
      <a>
        Compass Group
      </a>
    </li>
    <li>
      <a>
        Bilbary
      </a>
    </li>
    <li>
      <a>
        Genes Reunited
      </a>
    </li>
    <li>
      <a>
        The National Lottery
      </a>
    </li>
    <li>
      <a>
        Age Concern
      </a>
    </li>
    <li>
      Linklog
    </li>
  </ul>
</nav>
`;
